src/internal/functions/Add-DscResourceModule.ps1

```diff
@@ -0,0 +1,60 @@
+function Add-DscResourceModule {
+  <#
+  .SYNOPSIS
+    Vendor PowerShell modules into the build folder either by path or by retrieving the modules from the Gallery
+
+  .DESCRIPTION
+    Given a DSC Resource Info object, introspect on the source (if possible) to return information about
+    the resources parameters, including their help text, whether or not they are mandatory for calling the
+    resource with the Get method, and whether or not they are mandatory for calling the resource with the
+    Set method.
+
+    This information can be reliably retrieved if the DSC resource is implemented in PowerShell and is
+    not class based. Otherewise, best guesses are made about the mandatory status of the parameters via
+    the Required attribute and the help information is null.
+
+  .PARAMETER Name
+    Specifies the name of the module to save.
+
+  .PARAMETER Path
+    Specifies the location on the local computer to store the saved module.
+
+  .PARAMETER RequiredVersion
+    Specifies the exact version number of the module to save.
+    If left blank, will default to latest available.
+
+  .EXAMPLE
+    Add-DscResourceModule -TargetDir ./tmp -Name PowerShellGet -Version 2.2.3
+
+  #>
+  [CmdletBinding()]
+  param (
+    $Name,
+    $Path = './tmp',
+    $RequiredVersion
+  )
+
+  Begin { }
+
+  Process {
+    if (-not(Test-Path $Path)) {
+      if (-not(Test-Path $Path)) {
+        New-Item -Path $Path -Force -ItemType 'Directory'
+      }
+      $PathTmp = "$(Get-CanonicalPath($Path))_tmp"
```

Comment:
If you'd like to avoid having to define a function to test this functionality, you could instead use `Join-Path`:

```suggestion
$TemporaryFolderPath = Join-Path -Path $Path -ChildPath '_tmp'
```

This also automagically handles path separators for you.

Reply:
This would do exactly the wrong thing:
```
PS C:\vagrant> function Get-CanonicalPath($Path) { [System.IO.DirectoryInfo]::new($Path).Target } 
PS C:\vagrant> Join-Path -Path "C:\PerfLogs\" -ChildPath "_tmp" 
C:\PerfLogs\_tmp 
```
instead of
```
PS C:\vagrant> "$(Get-CanonicalPath("C:\PerfLogs\"))_tmp" 
C:\PerfLogs_tmp 
PS C:\vagrant>  
```
and the original code required `_tmp` to be a sibling, not a child of `$path`.

Reply:
☕ 😭 

If _all_ we want to do is trim the final slash, if one exists, we could use the `-replace` operand instead, I think:

```suggestion
      $PathTmp = $Path -Replace "(/|\\)$", '_tmp'
```

Reply:
using the `-replace`. let's see what the tests say.